src/components/EmbeddedWalletLayout/index.tsx

```diff
@@ -0,0 +1,57 @@
+import { Card } from "@stellar/design-system";
+import { useEffect, type ReactNode } from "react";
+
+import "./styles.scss";
+
+type EmbeddedWalletLayoutProps = {
+  organizationName: string;
+  organizationLogo?: string;
+  headerRight?: ReactNode;
+  showHeader?: boolean;
+  contentAlign?: "center" | "left";
+  children: ReactNode;
+};
+
+export const EmbeddedWalletLayout = ({
+  children,
+  organizationName,
+  organizationLogo,
+  headerRight,
+  showHeader = true,
+  contentAlign = "center",
+}: EmbeddedWalletLayoutProps) => {
+  useEffect(() => {
+    const previousTitle = document.title;
+    document.title = `${organizationName} Wallet`;
+    return () => {
+      document.title = previousTitle;
+    };
+  }, [organizationName]);
+
+  return (
+    <div className="EmbeddedWalletLayout">
+      {showHeader ? (
+        <header className="EmbeddedWalletHeader">
+          <div className="EmbeddedWalletHeader__brand">
+            {organizationLogo ? (
+              <img src={organizationLogo} alt={`${organizationName} logo`} />
+            ) : (
+              <span className="EmbeddedWalletBrand">{organizationName}</span>
+            )}
+          </div>
+          {headerRight ? <h2 className="EmbeddedWalletHeader__title">{headerRight}</h2> : null}
```

Comment:
Semantically, `h2` should follow `h1`, so let's use a generic `div` here.

```suggestion
          {headerRight ? <div className="EmbeddedWalletHeader__title">{headerRight}</div> : null}
```